src/ferp/util/compareArrays.js

```diff
@@ -0,0 +1,40 @@
+const MAX_OBJECT_DEPTH = 2;
+
+const shallowCompareObjects = (obj1, obj2, depth) => {
+  if (depth > MAX_OBJECT_DEPTH) {
+    throw new Error('Ferp does not support object comparison deeper than 2. Consider storing the object for a fast object reference comparison instead.');
```

Comment:
I think throwing an exception is super heavy handed.
```suggestion
    console.warn('Ferp does not support object comparison deeper than 2. Consider storing the object for a fast object reference comparison instead.');
    return false;
```